### DISCUSSION TITLE
Deploy and execute

### DISCUSSION THREAD

**Discussion by marcelomorgado** [PaulRBerg/prb-proxy#25] [Ideas]
Hi, nice project!

I'm wondering if wouldn't be useful to have some kind of `Factory.deployAndCall()` function where the user can send a transaction when he wants to execute some call just after the deployment (same tx). I'm thinking on on-boarding like processes.
Maybe it's a silly question, but do you see any security concern on doing so?

Thanks

**Comment by PaulRBerg** [comment]
Hey, glad you like it!

Hmm, sounds interesting. I can see the usefulness of such a method, but at the same I think that thanks to CREATE2 the process of interacting with a not-yet-deployed proxy has been simplified.

A frontend app could ask the user to immediately sign a transaction to the `execute` function on the proxy, knowing that there is no risk of reorgs with PRBProxy.

I have to mull this idea for a while!

**Reply by marcelomorgado** [reply]
Hi, thanks. Note: I've edited my comment since maybe it wasn't too clear before.

**Reply by PaulRBerg** [reply]
Nah, it was clear. I just have to think if the gas cost of an additional function is worth it, given that sequential transactions are a solution already.

**Reply by SolcAndMe** [reply]
Hey, I would be for adding a method which deploys and then calls `execute` on the deployed proxy contract. The main drawback for a protocol / app to use proxy accounts for their users is user onboarding - in particular the very first interaction between the user and the app. Sequential transactions still lead to pretty janky UX - primarily in conveying to the user for why two transactions are being sent instead of just one and secondly it also is more expensive for the user because the base tx cost has to be paid twice. Adding that method to the factory leads to very minimal gas overhead.

**Reply by PaulRBerg** [reply]
Thanks for chiming in to share your feedback, @SolcAndMe.

I now agree that implementing this functionality would be useful, but I must confess that my availability is limited these days - I may be able to implement this feature if you could sponsor its development. There's a link to my [Gitcoin Grant](https://gitcoin.co/grants/1657/paulrberg-open-source-engineering) in this repository.

**Reply by SolcAndMe** [reply]
Awesome - we'll do!

**Comment by PaulRBerg** [comment]
This functionality (deploy + execute) is now fully supported by the factory and the registry.

Going to lock this discussion.

https://github.com/PaulRBerg/prb-proxy/commit/ce9d2983959f50577ee85b1a74f664bcd59403fd
